File: frontend/src/types/appointment.ts
import { User, Service, AppointmentStatus } from './index';

export interface Appointment {
  id: number;
  clientId: number;
  providerId: number;
  scheduledStartTime: string;
  scheduledEndTime: string;
  status: AppointmentStatus;
  totalPrice: number;
  createdAt: string;
  updatedAt: string;
  client?: User;
  provider?: User;
  services: AppointmentService[];
  notes?: string;
  review?: import('./index').Review; // Reference to review if appointment has been reviewed
}

export interface AppointmentService {
  id: number;
  appointmentId: number;
  serviceId: number;
  priceAtBooking: number;
  service: Service;
}

export interface CreateAppointmentRequest {
  providerId: number;
  serviceIds: number[];
  scheduledStartTime: string;
  notes?: string;
}

export interface TimeSlot {
  startTime: string;
  endTime: string;
  isAvailable: boolean;
  reason?: string; // e.g., "Already booked", "Outside business hours"
}

export interface AvailabilityRequest {
  providerId: number;
  date: string; // Format: YYYY-MM-DD
  serviceIds?: number[];
}

export interface UpdateAppointmentRequest {
  status?: AppointmentStatus;
  notes?: string;
  scheduledStartTime?: string;
}

File: backend/FYLA.Core/Entities/Appointment.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using FYLA.Core.Enums;

namespace FYLA.Core.Entities
{
  public class Appointment
  {
    [Key]
    public int Id { get; set; }

    [Required]
    public int ClientId { get; set; }

    [Required]
    public int ProviderId { get; set; }

    [Required]
    public DateTime ScheduledStartTime { get; set; }

    [Required]
    public DateTime ScheduledEndTime { get; set; }

    [Required]
    public AppointmentStatus Status { get; set; } = AppointmentStatus.Pending;

    [Column(TypeName = "decimal(10,2)")]
    public decimal? TotalPrice { get; set; }

    [StringLength(500)]
    public string? Notes { get; set; }

    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;

    // Navigation properties
    [ForeignKey(nameof(ClientId))]
    public virtual User Client { get; set; } = null!;

    [ForeignKey(nameof(ProviderId))]
    public virtual User Provider { get; set; } = null!;

    public virtual ICollection<AppointmentService> Services { get; set; } = new List<AppointmentService>();
    public virtual Review? Review { get; set; }
  }
}
